internal/manifest/ready_check.go

```diff
@@ -2,227 +2,67 @@ package manifest
 
 import (
 	"context"
-	"encoding/json"
-	"errors"
 	"fmt"
-	"strings"
-	"time"
 
 	apiappsv1 "k8s.io/api/apps/v1"
 	apicorev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	k8slabels "k8s.io/apimachinery/pkg/labels"
 	"k8s.io/cli-runtime/pkg/resource"
 	"k8s.io/kubectl/pkg/util/deployment"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/kyma-project/lifecycle-manager/api/shared"
-	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
 	declarativev2 "github.com/kyma-project/lifecycle-manager/internal/declarative/v2"
-	"github.com/kyma-project/lifecycle-manager/pkg/util"
 )
 
-const (
-	ToWarningDuration                = 5 * time.Minute
-	customResourceStatePath          = "status.state"
-	ModuleCRWithCustomCheckWarning   = "module CR state not found or given customStateCheck.jsonPath is not exists"
-	ModuleCRWithNoCustomCheckWarning = "module CR state not found"
-)
-
-// NewCustomResourceReadyCheck creates a readiness check that verifies that the Resource in the Manifest
+// NewDeploymentReadyCheck creates a readiness check that verifies that the Resource in the Manifest
 // returns the ready state, if not it returns not ready.
-func NewCustomResourceReadyCheck() *CustomResourceReadyCheck {
-	return &CustomResourceReadyCheck{}
+func NewDeploymentReadyCheck() *DeploymentReadyCheck {
+	return &DeploymentReadyCheck{}
 }
 
-type CustomResourceReadyCheck struct{}
-
-var (
-	ErrNotSupportedState    = errors.New("module CR state not support")
-	ErrRequiredStateMissing = errors.New("required Ready and Error state mapping are missing")
-)
+type DeploymentReadyCheck struct{}
 
-func (c *CustomResourceReadyCheck) Run(ctx context.Context,
+func (c *DeploymentReadyCheck) Run(ctx context.Context,
 	clnt declarativev2.Client,
-	obj declarativev2.Object,
 	resources []*resource.Info,
 ) (declarativev2.StateInfo, error) {
-	if !isDeploymentReady(clnt, resources) {
-		return declarativev2.StateInfo{
-			State: shared.StateProcessing,
-			Info:  "module operator deployment is not ready",
-		}, nil
-	}
-	manifest, ok := obj.(*v1beta2.Manifest)
-	if !ok {
-		return declarativev2.StateInfo{State: shared.StateError}, v1beta2.ErrTypeAssertManifest
-	}
-	if manifest.Spec.Resource == nil {
-		return declarativev2.StateInfo{State: shared.StateReady}, nil
-	}
-	moduleCR := manifest.Spec.Resource.DeepCopy()
-
-	err := clnt.Get(ctx, client.ObjectKeyFromObject(moduleCR), moduleCR)
-	if err != nil {
-		if util.IsNotFound(err) && !manifest.DeletionTimestamp.IsZero() {
-			return declarativev2.StateInfo{State: shared.StateDeleting}, nil
-		}
-		return declarativev2.StateInfo{State: shared.StateError}, fmt.Errorf("failed to fetch resource: %w", err)
-	}
-	return HandleState(manifest, moduleCR)
+	deploymentState := getDeploymentState(clnt, resources)
+	return declarativev2.StateInfo{State: deploymentState}, nil
```

Comment:
why always return a `nil` error?

Reply:
Because in this implementation for the interface there is no return Error needed, if an error appears then the State will be set to `Error`.
There is a second implementation for the `ReadyCheck` which has an error case and is needed for testing.
Check here: https://github.com/kyma-project/lifecycle-manager/blob/8766d1503157a6a05027266eb02c77bd388936a7/internal/declarative/v2/ready_check.go#L31

Reply:
then please just return `getDeploymentState(clnt, resources), nil`